frontend/packages/kubevirt-plugin/src/components/modals/vm-flavor-modal/vm-flavor-modal.tsx

```diff
@@ -0,0 +1,203 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { getVmTemplate, getResource } from 'kubevirt-web-ui-components';
+import {
+  HandlePromiseProps,
+  withHandlePromise,
+  Dropdown,
+  convertToBaseValue,
+  Firehose,
+} from '@console/internal/components/utils';
+import { TemplateModel } from '@console/internal/models';
+import {
+  createModalLauncher,
+  ModalTitle,
+  ModalBody,
+  ModalComponentProps,
+  ModalFooter,
+} from '@console/internal/components/factory';
+import { k8sPatch, TemplateKind } from '@console/internal/module/k8s';
+import { Form, FormGroup, TextInput } from '@patternfly/react-core';
+import { VMKind, VMLikeEntityKind } from '../../../types';
+import {
+  getFlavor,
+  getMemory,
+  getCPU,
+  getOperatingSystem,
+  getWorkloadProfile,
+  isVM,
+  vCPUCount,
+} from '../../../selectors/vm';
+import {
+  getTemplateFlavors,
+  getTemplates,
+  selectVM,
+} from '../../../selectors/vm-template/selectors';
+import { getUpdateFlavorPatches } from '../../../k8s/patches/vm/vm-patches';
+import { VirtualMachineModel } from '../../../models';
+import {
+  CUSTOM_FLAVOR,
+  NAMESPACE_OPENSHIFT,
+  TEMPLATE_TYPE_LABEL,
+  TEMPLATE_TYPE_BASE,
+} from '../../../constants';
+import './_vm-flavor-modal.scss';
+
+const MB = 1000 ** 2;
+
+const getId = (field: string) => `vm-flavor-modal-${field}`;
+const dehumanizeMemory = (memory?: string) => {
+  if (!memory) {
+    return null;
+  }
+
+  return convertToBaseValue(memory) / MB;
+};
+
+const getFlavors = (vm: VMLikeEntityKind, templates: TemplateKind[]) => {
```

Comment:
we dont we move this to selectors ? I think it has a potential to be used in other places too

Reply:
ok, moved